### DISCUSSION TITLE
Kotlin library build error, expected version is 1.5.1

### DISCUSSION THREAD

**Discussion by nabadeepbg** [shreyashsaitwal/rush-cli#52] [General] [answered]
I am trying to use some of latest version kotlin library with rush but we are facing this build error :

module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.7.1, expected version is 1.5.1.

Have any way to resolve this issue ?

**Comment by shreyashsaitwal** [comment, marked as answer]
Rush currently only supports v1.5.32 of Kotlin. However, you can manually upgrade it. Here's how:
1. Download latest kotlin-compiler.zip from [here](https://github.com/JetBrains/kotlin/releases).

2. Go to `$RUSH_DATA_DIR/tools/kotlinc` and replace it's contents with the ones from the above ZIP. On Windows, `$RUSH_DATA_DIR` is `%AppData%\rush`; on Linux, it's `$HOME/rush`.
3. Then, go to `$RUSH_DATA_DIR/tools/kotlinc/lib`, and then copy `annotations-13.0.jar`, `kotlin-stdlib.jar` and `kotlib-stdlib-sources.jar` from there to `$RUSH_DATA_DIR/dev-deps/kotlin`.
4. Done!
